src/useWindowSize.ts

```diff
@@ -3,11 +3,14 @@ import { useEffect } from 'react';
 import useRafState from './useRafState';
 import { isClient } from './util';
 
-const useWindowSize = (initialWidth = Infinity, initialHeight = Infinity) => {
-  const [state, setState] = useRafState<{ width: number; height: number }>({
-    width: isClient ? window.innerWidth : initialWidth,
-    height: isClient ? window.innerHeight : initialHeight,
-  });
+const useWindowSize = (initialWidth = Infinity, initialHeight = Infinity, callback?: CallableFunction) => {
+  const [state, setState] = useRafState<{ width: number; height: number }>(
+    {
+      width: isClient ? window.innerWidth : initialWidth,
+      height: isClient ? window.innerHeight : initialHeight,
+    },
+    callback
+  );
```

Comment:
Instead of adding `callback` to `useRafState` can we add here on the next line a `useEffect` block like this:

```ts
useEffect(() => {
  if (!callback) return;
  callback(state);
}, [state.width, state.height]);
```

Reply:
Good call, updated 😄